src/SymbolServer.jl

```diff
@@ -42,20 +93,33 @@ function getstore(ssi::SymbolServerInstance, environment_path::AbstractString, p
                     if manifest !== nothing
                         @debug "Downloading cache files for manifest at $(manifest_filename)."
                         to_download = collect(validate_disc_store(ssi.store_path, manifest))
-                        batches = Iterators.partition(to_download, max(1, floor(Int, length(to_download)÷50)))
-                        for (i, batch) in enumerate(batches)
-                            percentage = round(Int, 100*(i - 1)/length(batches))
-                            progress_callback !== nothing && progress_callback("Downloading caches...", percentage)
-                            @sync for pkg in batch
-                                @async begin
-                                    yield()
-                                    uuid = packageuuid(pkg)
-                                    get_file_from_cloud(manifest, uuid, environment_path, ssi.depot_path, ssi.store_path, download_dir, ssi.symbolcache_upstream)
-                                    yield()
+                        try
+                            remove_non_general_pkgs!(to_download)
+                        catch err
+                            # if any errors, err on the side of caution and mark all as private, and continue
+                            @error """
+                            Symbol cache downloading: Failed to identify which packages to omit based on the General registry.
+                            All packages will be processsed locally""" err
+                            empty!(to_download)
+                        end
+                        if !isempty(to_download)
+                            n_done = 0
+                            n_total = length(to_download)
+                            @sync for batch in Iterators.partition(to_download, 100) # 100 connections at a time
+                                for pkg in batch
```

Comment:
I think the sync has to go inside, so we do a whole batch of 100 before moving to the next, right?
```suggestion
                            for batch in Iterators.partition(to_download, 100) # 100 connections at a time
                                @sync for pkg in batch
```